packages/react-reconciler/src/ReactInternalTypes.js

```diff
@@ -314,6 +319,7 @@ export type Dispatcher = {|
     subscribe: MutableSourceSubscribeFn<Source, Snapshot>,
   ): Snapshot,
   useOpaqueIdentifier(): any,
+  useRefresh?: () => <T>(?() => T, ?T) => void,
```

Comment:
This optional hook is tricky, since it prevents Flow from calling out this incompatibility:
https://github.com/facebook/react/blob/070372cde2d8b344a96ec63c92b5dcbec863073f/packages/react-debug-tools/src/ReactDebugHooks.js#L305-L322

If DevTools were used to inspect any component that used this new hook, it would cause a runtime error.

I'll add a follow up PR.

Reply:
#20460

Reply:
getCacheForType is also optional and there was some discussion for why there.

Reply:
Yeah I'm cool with it always being there